lib/ecto/schema.ex

```diff
@@ -2038,6 +2048,15 @@ defmodule Ecto.Schema do
     end
   end
 
+  defp parameterized_type?({type, opts}) when is_atom(type) and is_list(opts) do
+    Code.ensure_compiled(type) == {:module, type} &&
```

Comment:
A more reliable check is probably to use `Ecto.Type.primitive?` to see if it is a primitive type. If it is not a primitive type (and it is a tuple), then it is a parameterized type. Something like:

```elixir
  defp parameterized_type?({type, opts}) when is_atom(type) do
    Ecto.Type.primitive_type?({type, opts})
  end
```